testsuite/replay-verify/main.py

```diff
@@ -0,0 +1,298 @@
+        set_env_var(
+            pod_manifest["spec"]["containers"][0],
+            PUSH_METRICS_ENDPOINT,
+            get_env_var(PUSH_METRICS_ENDPOINT, "http://localhost:9091"),
+        )
+        retries = 0
+        while retries <= MAX_RETRIES:
+            try:
+                retries += 1
+                response = api_instance.create_namespaced_pod(
+                    namespace=self.namespace, body=pod_manifest
+                )
+                logger.info(f"Created pod {self.name}")
+                return
+            except ApiException as e:
+                logger.warning(
+                    f"Retry {retries}/{MAX_RETRIES} for pod {self.name} failed: {e}"
+                )
+                time.sleep(RETRY_DELAY)
+
+    def get_pod_exit_code(self):
+        # Check the status of the pod containers
+        for container_status in self.status.status.container_statuses:
+            if container_status.state.terminated:
+                return container_status.state.terminated.exit_code
+        return None
+
+    def get_pod_status(self):
+        pod_status = self.client.read_namespaced_pod_status(
+            name=self.name, namespace=self.namespace
+        )
+        return pod_status
+
+    def get_humio_log_link(self):
+        # TODO: Implement this ref:get_humio_link_for_node_logs
+        return f"https://humio.com/search?query=namespace%3D%22{self.namespace}%22%20pod%3D%22{self.name}%22"
+
+
+class ReplayScheduler:
+    def __init__(
+        self,
+        id,
+        start_version,
+        end_version,
+        ranges_to_skip,
+        worker_cnt=NUM_OF_WORKERS,
+        range_size=RANGE_SIZE,
+        network=Network.TESETNET,
+    ):
+        config.load_kube_config()
+        self.client = client.CoreV1Api()
+        self.id = id
+        self.namespace = "default"
+        self.start_version = start_version
+        self.end_version = end_version
+        self.ranges_to_skip = ranges_to_skip
+        self.range_size = range_size
+        self.ranges_to_skip = ranges_to_skip
+        self.current_workers = [None] * worker_cnt
+        self.tasks = []
+        self.network = network
+        self.failed_workpod_logs = []
+        self.txn_mismatch_logs = []
+
+    def get_label(self):
+        return f"{self.id}-{self.network}"
+
+    # Create ranges consisting of start_version and end_version
+    def create_ranges(self):
+        ranges = []
+        current = self.start_version
+
+        # Sort ranges_to_skip to ensure we process them in order
+        sorted_skips = sorted(self.ranges_to_skip, key=lambda x: x[0])
+
+        while current < self.end_version:
+            # Check if current position falls within any skip range
+            should_skip = False
+            for skip_start, skip_end in sorted_skips:
+                if skip_start <= current < skip_end:
+                    current = skip_end  # Jump to end of skip range
+                    should_skip = True
+                    break
```

Comment:
nit: should pop things out of the list as you create ranges? it probably doesn't matter much but feels like a meaningful waste.